### DISCUSSION TITLE
Meeting Notes for GSoC (5/27/2024)

### DISCUSSION THREAD

**Discussion by JackyZzZz** [theochem/Selector#209] [General]
# Meeting Notes

**Date:** 5/27/2024
**Participants:** @FanwangM  @JackyZzZz

## Meeting Overview

1. **Where to put the code**
   - Fork the current repo and develop locally for now.

2. **Necessity to understand the detail of different algorithms**
   - High-level understanding should be enough for setting up the interface.
   - Should at least know the input and output format.

3. **Designing the interface**
   - Should code up a very simple working prototype before implementing in detail.
   - Output can just be the index(2D-matrix) for now. If time permits, we can add more visual outputs such as graphs.
   - Ultimate goal is to wrap every algorithm under the [methods](https://github.com/theochem/Selector/tree/main/selector/methods) folder for direct access in the interface.
   - One-time use application for the current stage, no database layer is needed.
   - Unit-test along the way will be very nice, but should not spend too much time before completing the main objective of the project.

## Next Steps

**To-Do Before Next Meeting:**
- Code up a simple working prototype for the interface.
- Set-up Github Action workflows.
- Connect the repo with HuggingFace for CI/CD by Github Actions.

**Comment by FanwangM** [comment]
Thanks for the notes.  @JackyZzZz 
Look forward to this exciting project!
